### PR TITLE
Validate model name conflicts in the deployment flow

### PR DESCRIPTION
**Pull request by huwshimi**
Fixes: #3133.

### CODE REVIEW

**Comment by huwshimi** [comment]
QA:
- start a new model and add an app
- open the deployment flow
- enter a model name that is the same as one of your existing models and blur the field
- there should be a validation message
- enter a unique model name and blur the field
- there should be no validation message

**Comment by huwshimi** [comment]
[Check for model name conflicts](https://trello.com/c/1fUIgPnM/376-check-for-model-name-conflicts)

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-gce.jujugui.org:8080//job/juju-gui/8553/

File: jujugui/static/gui/src/app/init/component-renderers-mixin.js [hunk 1]
```diff
@@ -1017,6 +1017,7 @@ Browser: ${navigator.userAgent}`
         gtmEnabled={this.applicationConfig.GTM_enabled}
         listBudgets={this.plans.listBudgets.bind(this.plans)}
         listClouds={controllerAPI.listClouds.bind(controllerAPI)}
+        listModelsWithInfo={this._bound.listModelsWithInfo}
```
Comment (hatched) [inline jujugui/static/gui/src/app/init/component-renderers-mixin.js]: We probably don't need the model info for this.

File: jujugui/static/gui/src/app/components/deployment-flow/model-name/model-name.js [hunk 2]
```diff
@@ -7,12 +7,29 @@ const React = require('react');
...
     if (modelName) {
       modelName.focus();
     }
+    this._getModels();
+  }
+
+  /**
+    Get the user's models
+  */
+  _getModels() {
```
Comment (hatched) [inline jujugui/static/gui/src/app/components/deployment-flow/model-name/model-name.js]: Shouldn't this fetch the list of names on blur as well?

**Comment by huwshimi** [comment]
I'm going to close this because you're probably right, but that's a bigger issue that needs to be resolved in the future.

**Pull request closed without merging**